### PR TITLE
BFB-173: Create FMOD samples

### PR DESCRIPTION
**Pull request by unoctium1**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-173

**Description**
Create a scene demonstrating fmod samples. Includes a script showing how to play audio and modify parameters, as well as a audio source that plays on start and an example of timeline integration. Also includes an "Advanced Samples" button that directs users to [Fmod's scripting examples](https://www.fmod.com/resources/documentation-unity?version=2.02&page=examples.html)

**Screenshots / GIFs**
image

### CODE REVIEW

File: Assets/FMOD Samples/Scripts/FMODBasicSamples.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            //RuntimeManager.PlayOneShot(); 
+            // This is the FMOD equivalent of giving a script an audio clip and then playing it using an audio manager instead of a direct audio source,
+            // but it really complicates parameters
+        }
+        
+        public void PlayWithPitch()
+        {
+            var pitch = _pitchSlider.value;
+            _emitter.Play(); // note - play clears cached parameter values. Call this, THEN set params!
+            _emitter.SetParameter(PARAM_NAME, pitch);
```
Comment (m4rtint) [inline Assets/FMOD Samples/Scripts/FMODBasicSamples.cs:37]: so is this parameter set for the _emitter above?

Comment (unoctium1) [reply Assets/FMOD Samples/Scripts/FMODBasicSamples.cs:37]: It has a default value of 1 set in the inspector but otherwise its just getting set here

**Review by m4rtint — CHANGES_REQUESTED** [review]
The build currently breaks, so i'm going to wait until that's resolved before merging this in

File: Assets/FMOD Samples/Scripts/FMODBasicSamples.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Perigon.FMODExamples
+{
+    public class FMODBasicSamples : MonoBehaviour
+    {
+        private const string PARAM_NAME = "Pitch";
+        
+        [SerializeField] Slider _pitchSlider;
```
Comment (Gursaanj) [inline Assets/FMOD Samples/Scripts/FMODBasicSamples.cs:12]: should have `private` access modifier

**Review by m4rtint — APPROVED** [review]

**Pull request merged**